src/Shared/ParameterBindingMethodCache.cs

```diff
@@ -272,6 +279,27 @@ static bool ValidateReturnType(MethodInfo methodInfo)
         }
     }
 
+    private static MethodInfo? GetIBindableFromHttpContextMethod(Type type)
+    {
+        // Check if parameter is bindable via static abstract method on IBindableFromHttpContext<TSelf>
+        foreach (var i in type.GetInterfaces())
+        {
+            if (i.IsGenericType && i.GetGenericTypeDefinition() == typeof(IBindableFromHttpContext<>) && i.GetGenericArguments()[0] == type)
+            {
+                return BindAsyncMethod.MakeGenericMethod(type);
+            }
+        }
+
+        return null;
+    }
+
+    private static ValueTask<TValue?> BindAsync<TValue>(HttpContext httpContext, ParameterInfo parameter)
+        where TValue : class?, IBindableFromHttpContext<TValue>
+    {
+        return TValue.BindAsync(httpContext, parameter);
+    }
+        
+
```

Comment:
Nit:

```suggestion
```

I thought analyzers blocked the build for multiple newlines these days.